examples_tests/ComputePointCloud/main.cpp

```diff
@@ -0,0 +1,516 @@
+			}
+			m_visbuffer = logicalDevice->createDeviceLocalGPUImageOnDedMem(std::move(imageParams));
+
+			video::IGPUImageView::SCreationParams imgViewInfo;
+			{
+				imgViewInfo.flags = static_cast<video::IGPUImageView::E_CREATE_FLAGS>(0u);
+				imgViewInfo.image = m_visbuffer;
+				imgViewInfo.viewType = video::IGPUImageView::ET_2D;
+				imgViewInfo.format = asset::E_FORMAT::EF_R32_UINT;
+				imgViewInfo.subresourceRange.aspectMask = static_cast<asset::IImage::E_ASPECT_FLAGS>(0u);
+				imgViewInfo.subresourceRange.baseMipLevel = 0;
+				imgViewInfo.subresourceRange.levelCount = 1;
+				imgViewInfo.subresourceRange.baseArrayLayer = 0;
+				imgViewInfo.subresourceRange.layerCount = 1;
+			}
+			m_visbufferView = logicalDevice->createGPUImageView(std::move(imgViewInfo));
+		}
+
+		// Load the mesh
+		auto loadAndGetCpuMesh = [&](system::path path) -> std::pair<core::smart_refctd_ptr<asset::ICPUMesh>, const asset::IAssetMetadata*>
+		{
+			auto meshes_bundle = assetManager->getAsset(path.string(), {});
+			{
+				bool status = !meshes_bundle.getContents().empty();
+				assert(status);
+			}
+
+			auto mesh = core::smart_refctd_ptr_static_cast<asset::ICPUMesh>(meshes_bundle.getContents().begin()[0]);
+			auto metadata = meshes_bundle.getMetadata();
+			return std::make_pair(mesh, metadata);
+			//return std::make_pair(core::smart_refctd_ptr_static_cast<asset::ICPUMesh>(meshes_bundle.getContents().begin()[0]), meshes_bundle.getMetadata());
+		};
+
+		auto cpuBundlePLYData = loadAndGetCpuMesh(sharedInputCWD / "ply/Spanner-ply.ply");
+
+		core::smart_refctd_ptr<asset::ICPUMesh> cpuMeshPly = cpuBundlePLYData.first;
+		auto metadataPly = cpuBundlePLYData.second->selfCast<const asset::CPLYMetadata>();
+
+		// Get the positions from the mesh
+		uint32_t positionVbOffset;
+		core::smart_refctd_ptr<video::IGPUBuffer> positionVb;
+
+		{
+			const asset::ICPUMeshBuffer* const firstMeshBuffer = cpuMeshPly->getMeshBuffers().begin()[0];
+			const asset::SBufferBinding<const asset::ICPUBuffer> posBufferBinding = firstMeshBuffer->getVertexBufferBindings()[firstMeshBuffer->getPositionAttributeIx()];
+			core::smart_refctd_ptr<const asset::ICPUBuffer> posBuffer = posBufferBinding.buffer;
+
+			//const float* positions = static_cast<const float*>(posBuffer->getPointer());
+			//logger->log("Mesh buffer %i bytes %i", system::ILogger::E_LOG_LEVEL::ELL_INFO, 0, posBuffer->getSize());
+			//for (uint32_t i = 0; i < posBuffer->getSize() / 12; i++) {
+			//	logger->log("Pos %f %f %f", system::ILogger::E_LOG_LEVEL::ELL_INFO, positions[i * 3], positions[i * 3 + 1], positions[i * 3 + 2]);
+			//}
+
+			m_pointCount = posBuffer->getSize() / 12;
+			auto posBuffer_cpu = posBuffer.get();
+			auto positionsVertexBuffer = cpu2gpu.getGPUObjectsFromAssets(&posBuffer_cpu, &posBuffer_cpu + 1, cpu2gpuParams)->front();
+			positionVbOffset = positionsVertexBuffer->getOffset();
+			positionVb = core::smart_refctd_ptr<video::IGPUBuffer>(positionsVertexBuffer->getBuffer());
+		}
```

Comment:
MeshManipulator is more robust/resilient to de-interleaving different formats

Reply:
Also you have no guarantee that vertex buffer offset is okay to use with SSBO binding alignment, so I'd always repack